clippy_lints/src/deprecated_lints.rs

```diff
@@ -14,6 +14,16 @@ declare_deprecated_lint! {
     "`.extend_from_slice(_)` is a faster way to extend a Vec by a slice"
 }
 
+/// **What it does:** Nothing. This lint has been deprecated.
+///
+/// **Deprecation reason:** `Range::step_by(0)` used to be linted since it's
+/// an infinite iterator, which is better expressed by `iter::repeat`,
+/// but the method has been removed for `Iterator::step_by` which panics
+/// if given a zero
```

Comment:
Sound like we'd rather deprecate `RANGE_STEP_BY_ZERO` and introduce `ITERATOR_STEP_BY_ZERO` which basically does the same as the current lint.